File: testdata/sample.go
package sample

import (
	"context"
	"fmt"
	"io"
	"net/http"
)

// MaxRetries はリトライ最大回数。
const MaxRetries = 3

var DefaultTimeout = 30

// Status はステータスを表す列挙型。
type Status int

const (
	StatusPending Status = iota
	StatusActive
	StatusDone
)

// Config はアプリケーション設定。
type Config struct {
	Host    string
	Port    int
	Debug   bool
	handler http.Handler // unexported
}

// Server はHTTPサーバーを表す。
type Server struct {
	config Config
	mux    *http.ServeMux
}

// NewServer は Server を生成する。
func NewServer(cfg Config) *Server {
	return &Server{config: cfg, mux: http.NewServeMux()}
}

// Start はサーバーを起動する。
func (s *Server) Start(ctx context.Context) error {
	addr := fmt.Sprintf("%s:%d", s.config.Host, s.config.Port)
	return http.ListenAndServe(addr, s.mux)
}

// Shutdown はサーバーを停止する。
func (s *Server) Shutdown(ctx context.Context) error {
	return nil
}

// Handler はリクエストを処理するインターフェース。
type Handler interface {
	Handle(ctx context.Context, r *http.Request) error
	Close() error
}

// Reader はデータを読み込むインターフェース。
type Reader interface {
	io.Reader
	ReadAt(p []byte, off int64) (n int, err error)
}

// helper は非公開関数。
func helper() {}

// FormatStatus はステータスを文字列に変換する。
func FormatStatus(s Status) string {
	switch s {
	case StatusPending:
		return "pending"
	case StatusActive:
		return "active"
	case StatusDone:
		return "done"
	default:
		return "unknown"
	}
}

// Middleware は HTTP ミドルウェアの型エイリアス。
type Middleware = func(http.Handler) http.Handler

// HandlerFunc はハンドラ関数型。
type HandlerFunc func(ctx context.Context, w http.ResponseWriter, r *http.Request) error

File: testdata/sample.c
#include <stdio.h>
#include <stdlib.h>
#include "myheader.h"

#define MAX_RETRIES 3
#define DEFAULT_TIMEOUT 30
#define SQUARE(x) ((x) * (x))

typedef int Status;
typedef struct Node Node;
typedef void (*CallbackFunc)(int code, const char *msg);

enum LogLevel {
    LOG_DEBUG,
    LOG_INFO,
    LOG_WARN,
    LOG_ERROR
};

struct Config {
    const char *host;
    int port;
    int debug;
};

struct Server {
    struct Config config;
    int running;
};

static int helper_count = 0;

int global_timeout = 60;

struct Server *server_new(struct Config cfg);

void server_start(struct Server *s);

static void helper(void);

int format_status(Status s) {
    switch (s) {
    case 0: return 0;
    default: return -1;
    }
}

struct Server *server_new(struct Config cfg) {
    struct Server *s = malloc(sizeof(struct Server));
    s->config = cfg;
    s->running = 0;
    return s;
}

void server_start(struct Server *s) {
    s->running = 1;
}

static void helper(void) {
    helper_count++;
}

void server_free(struct Server *s) {
    free(s);
}
